lib/autoupdate/agent/updater.go

```diff
@@ -790,6 +765,50 @@ func (u *Updater) update(ctx context.Context, cfg *UpdateConfig, target Revision
 	}))
 }
 
+// Setup writes updater configuration and verifies the Teleport installation.
+// If restart is true, Setup also restarts Teleport.
+// Setup is safe to run concurrently with other Updater commands.
+func (u *Updater) Setup(ctx context.Context, restart bool) error {
+	// Setup teleport-updater configuration and sync systemd.
+
```

Comment:
I was tempted to include the linking / reverting logic here as well, but I think it introduces complexity that may be undesirable:
- The `LocalInstaller` logic would be split between versions, which may be difficult to reason about for future changes. E.g., we would not be able to change the format of the installation within a single version.
- Reverting logic would move to the new version, and never execute until it's installed.